src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_mask.c

```diff
@@ -1,59 +1,808 @@
+        false, // SLOT_MASK_BLAST
+        false, // SLOT_MASK_STONE
+        false, // SLOT_MASK_GREAT_FAIRY
+        true,  // SLOT_MASK_DEKU
+        false, // SLOT_MASK_KEATON
+        false, // SLOT_MASK_BREMEN
+        false, // SLOT_MASK_BUNNY
+        false, // SLOT_MASK_DON_GERO
+        false, // SLOT_MASK_SCENTS
+        true,  // SLOT_MASK_GORON
+        false, // SLOT_MASK_ROMANI
+        false, // SLOT_MASK_CIRCUS_LEADER
+        false, // SLOT_MASK_KAFEIS_MASK
+        false, // SLOT_MASK_COUPLE
+        false, // SLOT_MASK_TRUTH
+        true,  // SLOT_MASK_ZORA
+        false, // SLOT_MASK_KAMARO
+        false, // SLOT_MASK_GIBDO
+        false, // SLOT_MASK_GARO
+        false, // SLOT_MASK_CAPTAIN
+        false, // SLOT_MASK_GIANT
+        true,  // SLOT_MASK_FIERCE_DEITY
+    },
+    // Deku
+    {
+        false, // SLOT_MASK_POSTMAN
+        false, // SLOT_MASK_ALL_NIGHT
+        false, // SLOT_MASK_BLAST
+        false, // SLOT_MASK_STONE
+        false, // SLOT_MASK_GREAT_FAIRY
+        true,  // SLOT_MASK_DEKU
+        false, // SLOT_MASK_KEATON
+        false, // SLOT_MASK_BREMEN
+        false, // SLOT_MASK_BUNNY
+        false, // SLOT_MASK_DON_GERO
+        false, // SLOT_MASK_SCENTS
+        true,  // SLOT_MASK_GORON
+        false, // SLOT_MASK_ROMANI
+        false, // SLOT_MASK_CIRCUS_LEADER
+        false, // SLOT_MASK_KAFEIS_MASK
+        false, // SLOT_MASK_COUPLE
+        false, // SLOT_MASK_TRUTH
+        true,  // SLOT_MASK_ZORA
+        false, // SLOT_MASK_KAMARO
+        false, // SLOT_MASK_GIBDO
+        false, // SLOT_MASK_GARO
+        false, // SLOT_MASK_CAPTAIN
+        false, // SLOT_MASK_GIANT
+        true,  // SLOT_MASK_FIERCE_DEITY
+    },
+    // Human
+    {
+        true, // SLOT_MASK_POSTMAN
+        true, // SLOT_MASK_ALL_NIGHT
+        true, // SLOT_MASK_BLAST
+        true, // SLOT_MASK_STONE
+        true, // SLOT_MASK_GREAT_FAIRY
+        true, // SLOT_MASK_DEKU
+        true, // SLOT_MASK_KEATON
+        true, // SLOT_MASK_BREMEN
+        true, // SLOT_MASK_BUNNY
+        true, // SLOT_MASK_DON_GERO
+        true, // SLOT_MASK_SCENTS
+        true, // SLOT_MASK_GORON
+        true, // SLOT_MASK_ROMANI
+        true, // SLOT_MASK_CIRCUS_LEADER
+        true, // SLOT_MASK_KAFEIS_MASK
+        true, // SLOT_MASK_COUPLE
+        true, // SLOT_MASK_TRUTH
+        true, // SLOT_MASK_ZORA
+        true, // SLOT_MASK_KAMARO
+        true, // SLOT_MASK_GIBDO
+        true, // SLOT_MASK_GARO
+        true, // SLOT_MASK_CAPTAIN
+        true, // SLOT_MASK_GIANT
+        true, // SLOT_MASK_FIERCE_DEITY
+    },
 };
 
-s32 D_8082B684[] = {
-    0x01010004, 0x02020180, 0x01040210, 0x00100201, 0x00080110, 0x02040220, 0x00400080, 0x00020102, 0x00010240,
-    0x01200108, 0x00200140, 0x02080280, 0x00FF0064, 0x00FF0000, 0x00000064, 0x00FF0000, 0x000000FF, 0x00640000,
+#define SET_MOON_MASK_BIT(moonMasksBitsIndex, moonMasksBitsFlag) ((moonMasksBitsIndex) << 8 | (moonMasksBitsFlag))
```

Comment:
Is there any overlap between this use and what's happening in EnJs?

Reply:
They're checking the same thing, but doing so in different ways. Here, the index and flag are bitpacked together, while EnJs seems to keep the index and flag separate.

I was also thinking of something like `masksTakenOnMoonBits` to be a bit more verbose but idk

Reply:
I think it'd be okay since it's only used in a few files.